thread_primitives.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2012-2013, Aptarism SA.
```

Comment:
well this file is yours... you can update the copyright, and put your name there if you want :-)